apisix/plugins/authz-keycloak.lua

```diff
@@ -53,10 +54,26 @@ local schema = {
         keepalive_timeout = {type = "integer", minimum = 1000, default = 60000},
         keepalive_pool = {type = "integer", minimum = 1, default = 5},
         ssl_verify = {type = "boolean", default = true},
+        client_id = {type = "string", minLength = 1, maxLength = 100},
+        audience = {type = "string", minLength = 1, maxLength = 100,
+                    description = "Deprecated, use `client_id` instead."},
+        client_secret = {type = "string", minLength = 1, maxLength = 100},
+        lazy_load_paths = {type = "boolean", default = false},
+        http_method_as_scope = {type = "boolean", default = false},
+        cache_ttl_seconds = {type = "integer", minimum = 1, default = 24 * 60 * 60},
     },
     anyOf = {
         {required = {"discovery"}},
-        {required = {"token_endpoint"}}}
+        {required = {"token_endpoint"}}
+    },
+    dependencies = {
+        lazy_load_paths = {
```

Comment:
Should we not give a default value to `lazy_load_paths`?
Currently either discovery or resource_registration_endpoint is required as there is always a value for lazy_load_paths.

Reply:
Fixing. Defaults to `false`. Should require one of `discovery` or `resource_registration_endpoint` if set to `true`.

Reply:
Should be fixed now.